src/index/SortedIdTableMerge.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Felix Meisen (fesemeisen@outlook.de)
+
+#include "index/SortedIdTableMerge.h"
+
+#include "util/ParallelMultiwayMerge.h"
+
+namespace SortedIdTableMerge {
+
+// _____________________________________________________________________________
+IdTable mergeIdTables(
+    std::vector<IdTable>&& tablesToMerge,
+    const ad_utility::AllocatorWithLimit<ValueId>& allocator,
+    const ad_utility::MemorySize& memory, bool distinct,
+    const std::function<bool(const std::vector<ValueId>&,
+                             const std::vector<ValueId>&)>& comparator) {
+  std::vector<cppcoro::generator<std::vector<ValueId>>> generators;
+  auto makeGenerator =
+      [](const IdTable& table) -> cppcoro::generator<std::vector<ValueId>> {
+    for (const auto& row : table) {
+      co_yield std::vector<ValueId>(row.begin(), row.end());
+    }
+  };
```

Comment:
This is quite inefficient. 
Can't you just use the `IdTable::row_reference` type? 
And typically we should also use some `CALL_FIXED_SIZE` mechanism etc. 
The underlying problem is that this is a columnar data structure, and the algorithm we are using here is inherently row based.